### PR TITLE
FEATURE: JavaScript evaluation tool

### PR DESCRIPTION
**Pull request by SamSaffron**
This is similar to code interpreter by ChatGPT, except that it uses
JavaScript as the execution engine.

Safeguards were added to ensure memory is constrained and evaluation
times out.

### CODE REVIEW

**Comment by SamSaffron** [comment]
For example: 

image

File: lib/ai_bot/tools/javascript_evaluator.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+            #{script}
+          JS
+
+          result = context.eval(eval_script)
+
+          # only do special handling and truncating for long strings
+          if result.to_s.length > 1000
+            result = result.to_s
+            result = truncate(result.to_s, max_length: 10_000, percent_length: 0.3, llm: llm)
```
Comment (Drenmi) [inline lib/ai_bot/tools/javascript_evaluator.rb:76]: Seems like we're doubling up `result.to_s` here.

**Review by Drenmi — APPROVED** [review]
LGTM! 🤖

File: lib/ai_bot/tools/javascript_evaluator.rb (hunk 1 above)
Comment (SamSaffron) [reply lib/ai_bot/tools/javascript_evaluator.rb:76]: true , will fix

**Pull request merged**